modules/entity_to_text_tika/src/Storage/PlaintextStorage.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace Drupal\entity_to_text_tika\Storage;
+
+use Drupal\Core\File\FileSystemInterface;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Drupal\Core\StreamWrapper\StreamWrapperManager;
+use Drupal\Core\StreamWrapper\StreamWrapperManagerInterface;
+use Drupal\file\Entity\File;
+
+/**
+ * Provide Capabilities to store a Text content to plain-text file.
+ */
+class PlaintextStorage {
+
+  public const DESTINATION = 'private://entity-to-text/ocr';
+
+  /**
+   * The file system service.
+   *
+   * @var \Drupal\Core\File\FileSystemInterface
+   */
+  protected $fileSystem;
+
+  /**
+   * The logger service.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelInterface
+   */
+  protected $logger;
+
+  /**
+   * The stream wrapper manager.
+   *
+   * @var \Drupal\Core\StreamWrapper\StreamWrapperManagerInterface
+   */
+  protected $streamWrapperManager;
+
+  /**
+   * Construct a new PlaintextStorage object.
+   */
+  public function __construct(FileSystemInterface $file_system, LoggerChannelFactoryInterface $logger_factory, StreamWrapperManagerInterface $stream_wrapper_manager) {
+    $this->fileSystem = $file_system;
+    $this->logger = $logger_factory->get('entity_to_text_tika');
+    $this->streamWrapperManager = $stream_wrapper_manager;
+  }
+
+  /**
+   * Store a plain text value into a file.
+   *
+   * @param \Drupal\file\Entity\File $file
+   *   The document.
+   * @param string $langcode
+   *   The OCR langcode to be used.
+   *
+   * @return string|null
+   *   The transformed file into a plain text value by Apache Tika.
+   */
+  public function loadTextFromFile(File $file, string $langcode = 'eng'): ?string {
+    $fullpath = $this->getFullPath($file, $langcode);
+
+    if (!is_file($fullpath)) {
+      return NULL;
+    }
+
+    return file_get_contents($fullpath);
+  }
+
+  /**
+   * Store a plain text value into a file.
+   *
+   * @param \Drupal\file\Entity\File $file
+   *   The document to be saved.
+   * @param string $content
+   *   The plain-text document to be stored.
+   * @param string $langcode
+   *   The langcode.
+   *
+   * @return string
+   *   The saved fullpath file.
+   */
+  public function saveTextToFile(File $file, string $content, string $langcode = 'eng'): string {
+    $fullpath = $this->getFullPath($file, $langcode);
+    file_put_contents($fullpath, $content);
```

Comment:
Did you explored [`FileSystemInterface::saveData`](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21File%21FileSystemInterface.php/function/FileSystemInterface%3A%3AsaveData/10)? 

I didn't find the equivalent to read data, so I think I prefer to have a consistent approach with `file_puts_content()` and `file_get_contents()`